### DISCUSSION TITLE
How to split types in multiple files base on the tag ?

### DISCUSSION THREAD

**Discussion by quxiaodong** [acacode/swagger-typescript-api#1010] [Q&A]
I use the `npx swagger-typescript-api -p http://localhost:3000/swagger/json -o ./src/openapi -n index.ts --clean-output --no-client --module-name-index 1 --module-name-first-tag` to generate types

But all the types are in a same file, How can I do to split types in different files, such as `iam` and `administrator` ?

<img width="440" alt="image" src="https://github.com/user-attachments/assets/0dc5f351-a3c0-4bed-8fd2-c111edaafa93" />

**Comment by bks-abs** [comment]
i need this also . it is useful very much

**Comment by vamidi** [comment]
Would be really helpful, please implement 😄

**Comment by jorgen-appva** [comment]
```sh
# Answer: Generate modules per tag group. In this case for Axios. (works for `"openapi": "3.0.1"`)
npx swagger-typescript-api generate -p ./swagger.json -o ./openapi --module-name-first-tag --modular --axios --single-http-client
```

> What I need is confirmation of a planned road-map (an estimated date in the future) that this also officially will work for `"openapi": "3.1.x"`. Please help me bump #1516
